src/Generator.cpp

```diff
@@ -79,6 +79,11 @@ bool is_valid_name(const std::string &n) {
             return false;
         }
     }
+    // prohibit this specific string so that we can use it for
+    // passing GeneratorParams in Python.
+    if (n == "generator_params") {
```

Comment:
Alternately: instead of forbidding this name, we could make the optional argument to the call be `_generator_params` (ie, start with an underscore), since it has always been forbidden to start a Generator param, input, or output with an underscore (and thus no possibility of name collision). Not sure if this is considered to be a good thing in Python or not though?

Reply:
I prefer forbidding this name. Named arguments in Python rarely (never?) start with an underscore.

Reply:
Agreed, though `namedtuple` does make liberal use of starts-with-underscore names for various methods and fields that are definitely intended to be public -- wasn't sure if a similar exception might be appropriate here.